modules/lnd/lnd_wrapper/wrapper.go

```
@@ -162,9 +162,45 @@ func LndParseP2pLightningMessage(data *C.char, length C.int) *C.char {
 	case 19:
 		sb.WriteString("MSG_TYPE=pong;IGNORED=")
 		sb.WriteString(fmt.Sprintf("%d", len(message.(*lnwire.Pong).PongBytes)))
+	case 34:
+		fc := message.(*lnwire.FundingCreated)
+		// LND doesn't check the signature when parsing the message,
+		// but we do, otherwise the fuzzer will crash all the time.
+		err := validateSignature(fc.CommitSig)
+		if err != nil {
+			return C.CString("")
+		}
+
+		// Skip messages that have extra data. (more than 132 bytes)
+		// Since rust-lightning returns an error for messages that are too big.
+		if fc.ExtraData != nil {
+			return nil
+		}
+		sb.WriteString("MSG_TYPE=funding_created;TEMPORARY_CHANNEL_ID=")
+		sb.WriteString(fmt.Sprintf("%x", fc.PendingChannelID))
+		sb.WriteString(";FUNDING_TXID=")
+		sb.WriteString(fc.FundingPoint.Hash.String())
+		sb.WriteString(";FUNDING_OUTPUT_INDEX=")
+		sb.WriteString(fmt.Sprintf("%d", fc.FundingPoint.Index))
+		sb.WriteString(";SIGNATURE=")
+		sb.WriteString(fmt.Sprintf("%x", fc.CommitSig.ToSignatureBytes()))
 	}
 
 	return C.CString(sb.String())
 }
 
+func validateSignature(sig lnwire.Sig) error {
+	_, err := sig.ToSignature()
+	if err != nil {
+		// Both rust-lightning and c-lightning checks if R and S are 0 when
+		// verifying the signature against the public key.
+		if err.Error() == "invalid signature: R is 0" ||
+			err.Error() == "invalid signature: S is 0" {
+			return nil
```

This ignores the errors for zero-value R or S.  But what if there are also other things wrong with the signature, but the zero-value R or S was detected first?  I'm afraid we may let the invalid signature through in that case, which could cause false positives when comparing to CLN or LDK.

Is it possible to *add* the full R/S validation to CLN and LDK instead?

---

True. Fixed